.github/workflows/coverage.yml

```diff
@@ -0,0 +1,44 @@
+name: Code coverage
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Install lcov
+      shell: bash
+      run: |
+          echo "::set-env name=CC::gcc-8"
+          echo "::set-env name=CXX::g++-8"
+
+    - name: Create Build Environment
+      shell: bash
+      run: cmake -E make_directory ${{runner.workspace}}/build
+
+    - name: Configure
+      shell: bash
+      working-directory: ${{runner.workspace}}/build
+      run: cmake $GITHUB_WORKSPACE -DCMAKE_BUILD_TYPE=Debug -DPISA_BUILD_TOOLS=OFF -DENABLE_COVERAGE=ON ..
+
+    - name: Build
+      shell: bash
+      working-directory: ${{runner.workspace}}/build
+      run: cmake --build . --config Debug -- -j 4
+
+    - name: Test
+      shell: bash
+      working-directory: ${{runner.workspace}}/build
+      run: ctest -j 4
+
+    - name: Upload results
+      shell: bash
+      working-directory: ${{runner.workspace}}/build
+      run: bash <(curl -s https://codecov.io/bash);
```

Comment:
You are missing the codecov token to submit new results. I will send it over once I find it :)

Reply:
CODECOV_TOKEN="e3f86b6b-0770-4816-9b7d-19b0c550229f"